support-lib/jni/djinni_support.cpp

```diff
@@ -239,8 +239,13 @@ jint JniEnum::ordinal(JNIEnv * env, jobject obj) const {
 
 LocalRef<jobject> JniEnum::create(JNIEnv * env, jint value) const {
     LocalRef<jobject> values(env, env->CallStaticObjectMethod(m_clazz.get(), m_staticmethValues));
+    jniExceptionCheck(env);
     DJINNI_ASSERT(values, env);
-    return LocalRef<jobject>(env, env->GetObjectArrayElement(static_cast<jobjectArray>(values.get()), value));
+    LocalRef<jobject> result(env,
+                             env->GetObjectArrayElement(static_cast<jobjectArray>(values.get()),
+                                                        value));
+    jniExceptionCheck(env);
+    return std::move(result);
```

Comment:
Overlapping with something with the same value wouldn't be allowed in general (same reason why the empty struct has size 1 - distinct objects have to have distinct addresses). My goto example for const-related UB is locating constants entirely in ROM, but that doesn't apply here.

The "treat x like a const reference to a non-const temporary" approach sounds like it could work, but there may be something I'm missing.